### PR TITLE
tools/package-management: replace name with pname&version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/package-management/disnix/dysnomia/default.nix [hunk 1]
```diff
@@ -38,10 +38,11 @@ assert enableNginxWebApplication -> nginx != null;
...
 
-stdenv.mkDerivation {
-  name = "dysnomia-0.10.1";
+stdenv.mkDerivation rec {
+  pname = "dysnomia";
+  version = "0.10.1";
   src = fetchurl {
-    url = "https://github.com/svanderburg/dysnomia/releases/download/dysnomia-0.10.1/dysnomia-0.10.1.tar.gz";
+    url = "https://github.com/svanderburg/dysnomia/releases/download/dysnomia-${version}/dysnomia-${version}.tar.gz";
     sha256 = "0w9601g8zpaxrmynx6mh8zz85ldpb8psp7cc6ls8v3srjpj1l5n3";
```
Comment (svanderburg) [inline pkgs/tools/package-management/disnix/dysnomia/default.nix:46]: maybe you can also use `${pname}` in the URL?

File: pkgs/tools/package-management/disnix/disnixos/default.nix [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
 
-stdenv.mkDerivation {
-  name = "disnixos-0.9.1";
+stdenv.mkDerivation rec {
+  pname = "disnixos";
+  version = "0.9.1";
 
   src = fetchurl {
-    url = "https://github.com/svanderburg/disnixos/releases/download/disnixos-0.9.1/disnixos-0.9.1.tar.gz";
+    url = "https://github.com/svanderburg/disnixos/releases/download/disnixos-${version}/disnixos-${version}.tar.gz";
```
Comment (svanderburg) [inline pkgs/tools/package-management/disnix/disnixos/default.nix:8]: aybe you can also use ${pname} in the URL?

File: pkgs/tools/package-management/disnix/default.nix [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 
-stdenv.mkDerivation {
-  name = "disnix-0.10.1";
+stdenv.mkDerivation rec {
+  pname = "disnix";
+  version = "0.10.1";
 
   src = fetchurl {
-    url = "https://github.com/svanderburg/disnix/releases/download/disnix-0.10.1/disnix-0.10.1.tar.gz";
+    url = "https://github.com/svanderburg/disnix/releases/download/disnix-${version}/disnix-${version}.tar.gz";
```
Comment (svanderburg) [inline pkgs/tools/package-management/disnix/default.nix:8]: aybe you can also use ${pname} in the URL?

File: pkgs/tools/package-management/disnix/dysnomia/default.nix (hunk 1 above)
Comment (Stunkymonkey) [reply pkgs/tools/package-management/disnix/dysnomia/default.nix:46]: i thought the only thing, that will change in the future is the whole URL or only the `version`. So yes i could change it, but i do not see any benefit from it. I even think this is easier to debug when updating.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131718` at 915e337f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>checkinstall</li>
  </ul>
</details>
<details>
  <summary>7 packages built successfully:</summary>
  <ul>
    <li>DisnixWebService</li>
    <li>disnix</li>
    <li>disnixos</li>
    <li>dydisnix</li>
    <li>dysnomia</li>
    <li>nix-serve</li>
    <li>opkg-utils</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/package-management/nix-serve/default.nix:32:3:
```
   |
32 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/tools/package-management/nix-serve/default.nix:46:5:
```
   |
46 |     license = licenses.lgpl21;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/package-management/opkg-utils/default.nix:21:5:
```
   |
21 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/tools/package-management/checkinstall/default.nix [hunk 4]
```diff
@@ -2,11 +2,12 @@
...
 
-stdenv.mkDerivation {
-  name = "checkinstall-1.6.2";
+stdenv.mkDerivation rec {
+  pname = "checkinstall";
+  version = "1.6.2";
 
   src = fetchurl {
-    url = "http://www.asic-linux.com.mx/~izto/checkinstall/files/source/checkinstall-1.6.2.tar.gz";
+    url = "http://www.asic-linux.com.mx/~izto/checkinstall/files/source/checkinstall-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/checkinstall/default.nix:10]:
```suggestion
    url = "https://www.asic-linux.com.mx/~izto/checkinstall/files/source/checkinstall-${version}.tar.gz";
```

File: pkgs/tools/package-management/disnix/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/tools/package-management/disnix/default.nix:8]: We shouldn't substitute pname everywhere. This makes pname normalization harder than necessary and is also coincidence.

**Pull request merged**